paymentsource/client.go

```diff
@@ -150,16 +150,20 @@ func Verify(id string, params *stripe.SourceVerifyParams) (*stripe.PaymentSource
 
 func (s Client) Verify(id string, params *stripe.SourceVerifyParams) (*stripe.PaymentSource, error) {
 	body := &stripe.RequestValues{}
-	body.Add("amounts[]", strconv.Itoa(int(params.Amounts[0])))
-	body.Add("amounts[]", strconv.Itoa(int(params.Amounts[1])))
 
 	source := &stripe.PaymentSource{}
 	var err error
 
 	if len(params.Customer) > 0 {
+		body.Add("amounts[]", strconv.Itoa(int(params.Amounts[0])))
+		body.Add("amounts[]", strconv.Itoa(int(params.Amounts[1])))
+
 		err = s.B.Call("POST", fmt.Sprintf("/customers/%v/sources/%v/verify", params.Customer, id), s.Key, body, &params.Params, source)
 	} else {
-		err = errors.New("Only customer bank accounts can be verified in this manner.")
+		body.Add("values[]", strconv.Itoa(int(params.Amounts[0])))
```

Comment:
If you look at the signature of SourceVerifyParams[0], it was really built for bank account verification on customer attached bank account objects (despite the correct intent of making it a generic name)

The right type for the payment agnostic `/v1/sources/:id/verify` endpoint is:
```
type NewNameOrMaybeSameNameSourceVerifyParams struct {
  Values []string
}
```

Unclear to me if we should rename the old one, or rename the new one, but these are two different constructs in the API. Hope that helps?

[0] https://github.com/stripe/stripe-go/blob/8c52b2ccae8b855e33ce2e1339646f54ec791b59/payment_source.go#L37
[1] https://github.com/stripe-internal/pay-server/blob/master/api/lib/api_method/source_verify.rb#L13